spring-boot-starter-camunda-test need jdk 21 instead 17

**Describe the bug**

https://github.com/camunda-community-hub/spring-zeebe/blob/main/test/embedded/pom.xml

use java version 21 instead 17

**To Reproduce**

`<dependency>
<groupId>io.camunda.spring</groupId>
<artifactId>spring-boot-starter-camunda-test</artifactId>
<scope>test</scope>
<version>8.4.0</version>
</dependency>`

`
@SpringBootTest
@ZeebeSpringTest
public class ZeebeCoreAppTest {

...`


**Expected behavior**

No Exception

**Log/Stacktrace**

```
[ERROR] /..../ZeebeCoreAppTest.java:[9,36] cannot access io.camunda.zeebe.spring.test.ZeebeSpringTest
  bad class file: ..../ZeebeSpringTest.class)
    class file has wrong version 65.0, should be 61.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
```

**Environment:**
- OS: Windows
- Deployment: -
- Spring-Zeebe Version: 8.4.0
- SpringBoot Version: 3.2.2
- Configuration: -

--- a/README.md
+++ b/README.md
@@ -272,11 +272,11 @@ You can start up an in-memory test engine and do assertions by adding this Maven
 </dependency>
 ```
 
-Note that **the test engines requires Java version >= 17**. If you cannot run on this Java version, you can use [Testcontainers](https://www.testcontainers.org/) **instead**. Testcontainers require that you have a Docker installation locally available on the developer machine. Use this dependency:
+Note that **the test engines requires Java version >= 21**. If you cannot run on this Java version, you can use [Testcontainers](https://www.testcontainers.org/) **instead**. Testcontainers require that you have a Docker installation locally available on the developer machine. Use this dependency:
 
 ```xml
 <!--
-  Alternative dependency if you cannot run Java 17, so you will leverage Testcontainer
+  Alternative dependency if you cannot run Java 21, so you will leverage Testcontainer
   Make sure NOT to have spring-zeebe-test on the classpath in parallel!
 -->
 <dependency>

